flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/source/abilities/SupportsRowLevelModificationScan.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.connector.source.abilities;
+
+import org.apache.flink.table.connector.RowLevelModificationScanContext;
+import org.apache.flink.table.connector.source.ScanTableSource;
+
+import javax.annotation.Nullable;
+
+/**
+ * Enables to tell the {@link ScanTableSource} the type of row-level modification the table source
+ * scan is for, and return {@link RowLevelModificationScanContext} which will then be passed to the
+ * sink which implements {@link SupportsRowLevelUpdate} or {@link SupportsRowLevelDelete}. It allows
+ * the table source to pass information to the table sink which is to be updated/deleted.
+ *
+ * <p>Note: This interface is optional for table source to support update/delete existing data. For
+ * the case that the table source won't need to know the type of row-level modification or pass
+ * information to sink, the table source doesn't need to implement it.
+ *
+ * <p>Note: Only if the {@link ScanTableSource} implements this interface, and the table is to be
+ * scanned to update/delete a table, the method {@link #applyRowLevelModificationScan} for the
+ * corresponding table source will be involved . For more details, please see the method {@link
+ * #applyRowLevelModificationScan}.
+ */
+public interface SupportsRowLevelModificationScan {
+
+    /**
+     * Apply the type of row-level modification and the previous {@link
+     * RowLevelModificationScanContext} returned by previous table source scan, return a new {@link
```

Comment:
Considering it maynot the final one the the table sink, I change it to 
`
If the table source is the last one, the {@link RowLevelModificationScanContext} will be passed to the table sink. Otherwise, it will be  passed to the following table source.
`